thrift/server.cc

```diff
@@ -165,7 +165,11 @@ thrift_server::connection::process_one_request() {
     };
     // Heuristics copied from transport/server.cc
     size_t mem_estimate = 8000 + 2 * _input->available_read();
-    auto units = co_await get_units(_server._memory_available, mem_estimate);
+    auto fut = get_units(_server._memory_available, mem_estimate);
+    if (_server._memory_available.waiters()) {
```

Comment:
Nitpick: Does waiters() mean that *we* will need to wait?
Wouldn't fut.available() be a more direct way to check if we need to wait or not?

Reply:
Yes, it's for checking if we will wait. I did it like that just because the original code in transport/server.cc is almost identical - and there's some value in keeping these two similar, since people familiar with the cql server code would immediately understand what's going on in here.